packages/back-end/src/api/snapshots/getExperimentSnapshot.ts

```diff
@@ -0,0 +1,32 @@
+import { findSnapshotById } from "../../models/ExperimentSnapshotModel";
+import { GetExperimentSnapshotResponse } from "../../../types/openapi";
+import { getExperimentById } from "../../models/ExperimentModel";
+import { createApiRequestHandler } from "../../util/handler";
+import { getExperimentSnapshotValidator } from "../../validators/openapi";
+
+export const getExperimentSnapshot = createApiRequestHandler(
+  getExperimentSnapshotValidator
+)(
+  async (req): Promise<GetExperimentSnapshotResponse> => {
+    const snapshot = await findSnapshotById(req.context.org.id, req.params.id);
+    if (!snapshot) {
+      throw new Error("Snapshot not found");
+    }
+    // no permission check in above method, so have to make sure they can read
+    // experiment first, which will be thrown by getExperimentById method
```

Comment:
Do we need to capture this for future work?

Reply:
maybe, but it's just replicating work elsewhere and I don't want to fix this permissions issue so we can get this out in 3.3